numpy/ma/core.py

```diff
@@ -3183,17 +3186,19 @@ def __setitem__(self, indx, value):
         locations.
 
         """
-        # 2016.01.15 -- v1.11.0
-        warnings.warn(
-           "Currently, slicing will try to return a view of the data," +
-           " but will return a copy of the mask. In the future, it will try" +
-           " to return both as views. This means that using `__setitem__`" +
-           " will propagate values back through all masks that are present.",
-           FutureWarning
-        )
-
         if self is masked:
             raise MaskError('Cannot alter the masked element.')
+
+        if (isinstance(self.mask, np.ndarray) and
```

Comment:
Oh, right, the logic was the other way around, it currently is always a copy and in the future always a view. There is a trick to do a warning on write stuff, but it is a bit ugly (and it has currently an error message combining diagonal and field access already.